packages/core/ios/src/Core/RCTAEPCore.m

```diff
@@ -166,6 +171,63 @@ - (NSData *)dataFromHexString:(NSString *)string {
      [AEPMobileCore resetIdentities];
 }
 
+
+/**
+ * Initializes the AEP Mobile SDK with the provided initialization options.
+ * @param {NSDictionary} initOptionsDict - The options to use for initialization.
+ *   - `appId` (NSString, required): A unique identifier assigned to the app instance by Adobe Experience Platform.
+ *   - `lifecycleAutomaticTrackingEnabled` (NSNumber, optional): Determines whether automatic lifecycle 
+ *      tracking should be enabled. Defaults to `true` if not provided.
+ *   - `lifecycleAdditionalContextData` (NSDictionary, optional): Key-value pairs of additional context data 
+ *      to be included with lifecycle events.
+ * @param {RCTPromiseResolveBlock} resolve - The promise resolve block.
+ * @param {RCTPromiseRejectBlock} reject - The promise reject block.
+ */
+RCT_EXPORT_METHOD(initialize:(NSDictionary *)initOptionsDict 
+                  resolver:(RCTPromiseResolveBlock)resolve 
+                  rejecter:(RCTPromiseRejectBlock)reject) {
+    
+    if (!initOptionsDict || ![initOptionsDict isKindOfClass:[NSDictionary class]]) {
+        reject(EXTENSION_NAME, @"InitOptions must be a valid dictionary.", nil);
+        return;
+    }
+
+    @try {
+        // Extract appId safely
+        NSString *appId = initOptionsDict[APP_ID_KEY];
+        if (![appId isKindOfClass:[NSString class]] || appId.length == 0) {
```

Comment:
AEPInitOptions *options;

if (![appID isKindOfClass:[NSString class]] || appID.length == 0) {
    options = [[AEPInitOptions alloc] init];
} else {
    options = [[AEPInitOptions alloc] initWithAppId:appID];
}

Reply:
The SDK will use bundled config if app id is not present.